File: Client/Game_FruitMachine/src/Dto/GameCommand/GameResultDto.ts
namespace Dto {
    /**
     * 游戏结果Dto
     */
    export class GameResultDto {
        /**
         * 游戏结果位置
         */
        public Result: Enum.ResultPosEnum;
        /**
         * 余额
         */
        public Balance: number;
        /**
         *  中奖金额 0为未中奖
         */
        public WinAmount: any;
        /**
         * 投注状态
         */
        public Status: Enum.BetResultCode;
        /**
         * 是否为猜大小
         */
        public GuessResult:boolean;
    }
}

File: QIC.EGame.GameMachineService/QIC.EGame.GameMachineService.GoalFootball/Dtos/GameResultDto.cs
using QIC.EGame.GameMachineService.GoalFootball.Enums;

namespace QIC.EGame.GameMachineService.GoalFootball.Dtos
{
    /// <summary>
    /// 注单结果的Dto
    /// </summary>
    public class GameResultDto
    {
        /// <summary>
        /// 新的余额
        /// </summary>
        public decimal Balance { get; set; }

        /// <summary>
        /// 中奖金额，0表示未中奖
        /// </summary>
        public decimal WinAmount { get; set; }

        /// <summary>
        /// 投注状态，指示投注是否成功
        /// </summary>
        public BetResult Status { get; set; }

        public static GameResultDto ParameterError
        {
            get
            {
                return new GameResultDto()
                {
                    Status = BetResult.ParameterError
                };
            }
        }
        public static GameResultDto InsufficientBalance
        {
            get
            {
                return new GameResultDto()
                {
                    Status = BetResult.InsufficientBalance
                };
            }
        }
        public static GameResultDto OverLimit
        {
            get
            {
                return new GameResultDto()
                {
                    Status = BetResult.OverLimit
                };
            }
        }
        public static GameResultDto SystemError
        {
            get
            {
                return new GameResultDto()
                {
                    Status = BetResult.SystemError
                };
            }
        }
    }
}